backend/user-service/src/middleware/authMiddleware.ts

```diff
@@ -0,0 +1,51 @@
+import { Request, Response, NextFunction } from "express";
+import jwt from "jsonwebtoken";
+// import "dotenv/config";
+
+interface User {
+  id: number;
+  username: string;
+  email: string;
+  role: string;
+  languages: string[];
+}
+
+interface JwtPayload {
+  user: User;
+  exp: number;
+  iat: number;
+}
+
+const ACCESS_TOKEN_SECRET = process.env.ACCESS_TOKEN_SECRET as string;
+const REFRESH_TOKEN_SECRET = process.env.REFRESH_TOKEN_SECRET as string;
+
+// basically verify access token
+export async function verifyAccessToken(req: Request, res: Response, next: NextFunction) {
+  const accessToken = req.cookies["accessToken"]; // If JWT token is stored in a cookie
+
+  if (!accessToken) {
+    res.status(401).json({ errors: [{ msg: 'Not authorized, no access token' }] });
+  } else {
+    jwt.verify(
+      accessToken, 
+      ACCESS_TOKEN_SECRET,
+      (err: Error | null) => {
+        if (err) {
+          res.status(401).json({ errors: [{msg: 'Not authorized, access token failed'}] });

```

Comment:
```suggestion
          res.status(401).json({ errors: [{msg: 'Not authorized, access token failed'}] });
          return;
```
then we would not need to check for things sent in following middleware (etc logout)
also causes /currentUser to crash on token expiry